src/convolver_ui.cpp

```diff
@@ -303,18 +303,18 @@ void get_irs_spectrum(ConvolverBox* self, const int& rate) {
     self->data->right_spectrum[n] = (self->data->right_spectrum[n] - fft_min_right) / (fft_max_right - fft_min_right);
   }
 
+  g_object_ref(self);
   util::idle_add([=]() {
-    if (self == nullptr) {
-      return;
-    }
-
-    if (!ui::chart::get_is_visible(self->chart)) {
-      return;
-    }
-
-    if (gtk_toggle_button_get_active(self->show_fft) != 0) {
+    
+    if (  
+      ( self != nullptr ) &&
+      ( ui::chart::get_is_visible(self->chart) ) &&
+      ( gtk_toggle_button_get_active(self->show_fft) != 0 )
+    ) {
```

Comment:
This check will crash if `self == nullptr` because the check `ui::chart::get_is_visible(self->chart)` will be executed anyway and `self->chart` will point to null. Same issue on the other check.

Reply:
Something like this will have to be done
```
if (self == nullptr) {
     g_object_unref(self);
      return;
  }

if (!ui::chart::get_is_visible(self->chart)) {
  g_object_unref(self);
  return;
}
```
and in the end the `g_object_unref(self);` you have already added. This one is executed in case the other checks fail and the callback code goes to the end.

Reply:
I thought conditions with an && operator are checked in order until one returns not true; so that in this case, if self == nullptr, evaluation would stop there, skip the rest of the checks, and only execute g_object_unref

Reply:
>I thought conditions with an && operator are checked in order until one returns not true;

I am almost 100% sure I tried this in the past and the code crashed. Or maybe at the time it was an `||` and not a `&&`. In any case it is probably better to play safe and keep them separated just to avoid being surprised by a different compiler behavior.

Reply:
[this so andwer](https://stackoverflow.com/questions/7925479/if-argument-evaluation-order) says that the left to right semantics of && is standardized

Reply:
Ok. I did a quick test and at least on gcc it seems to be the case.